aten/src/ATen/functorch/BatchedTensorImpl.cpp

```diff
@@ -61,6 +84,34 @@ void BatchedTensorImpl::checkInvariants() const {
   TORCH_INTERNAL_ASSERT(level_ > -1);
 }
 
+int64_t BatchedTensorImpl::size_custom(int64_t d) const {
+  if (!value_.is_nested()) {
+    return size(d);
+  }
+  // TODO: Error messages will mention the actualDim, which could be confusing; fix this
```

Comment:
Thoughts on this?

Reply:
Eh, not the worse thing in the world. We have this problem already with our batching rules.

Reply:
Sure:

```python
def f(x):
    x.size(0)
    return x

# Shape (3, *, 2) where * is ragged / irregular
x = self._create_nt([3, None, 2], device=device)

# Given dimension 1 is irregular and does not have a size.
vmap(f)(x)
```

So `f` calls `size(0)` but the error message mentions dimension 1. I think this is specific to NT since we error out for `size()` calls on irregular dims.